server/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -509,13 +512,23 @@ private static OrderBy analyzeOrderBy(SelectAnalysis selectAnalysis,
     private List<Symbol> analyzeGroupBy(SelectAnalysis selectAnalysis,
                                         List<Expression> groupBy,
                                         ExpressionAnalyzer expressionAnalyzer,
-                                        ExpressionAnalysisContext expressionAnalysisContext) {
+                                        ExpressionAnalysisContext expressionAnalysisContext,
+                                        boolean isGroupByAll) {
         List<Symbol> groupBySymbols = new ArrayList<>(groupBy.size());
-        for (Expression expression : groupBy) {
-            Symbol symbol = symbolFromExpressionFallbackOnSelectOutput(
-                expression, selectAnalysis, "GROUP BY", expressionAnalyzer, expressionAnalysisContext);
-            GroupBySymbolValidator.validate(symbol);
-            groupBySymbols.add(symbol);
+        if (groupBy.isEmpty() && isGroupByAll) {
+            for (Symbol symbol : selectAnalysis.outputSymbols()) {
+                if (!symbol.hasFunctionType(FunctionType.AGGREGATE) && !symbol.hasFunctionType(FunctionType.TABLE)) {
+                    GroupBySymbolValidator.validate(symbol);
+                    groupBySymbols.add(symbol);
```

Comment:
What happens if an aggregation is wrapped in a function? `select 1 + count(*), x from tbl group by all` ?

Reply:
Hello @mfussenegger 
Table 
![image](https://github.com/user-attachments/assets/4368f2b3-5e95-4e11-abd0-74248f4300ff)
Query - select 1 + count(*), department from employees group by all;
Result 
![image](https://github.com/user-attachments/assets/4d7bbb21-8612-4d74-b22f-da7f6c378667)


Please verify if the result is a expected

Reply:
That's correct :+1: